dependencies/CMakeLists.txt

```diff
@@ -22,11 +22,17 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-include(FetchContent)
 
```

Comment:
Are you sure this works? I'm not quite sure, but if I remember correctly find_package has to be called in the (parent) scope of where it's actually linked to. 

That means find_package would need to be in the root CMakeFile (just as in your previous version and the if else has to be around the add_subdirectory(dependencies). 

I'm not 100% sure though if I remember here correctly, I can't test it myself right now

Reply:
Yes, I checked that it works on my system (Arch Linux x86-64, cmake 3.21.1) in both configurations: with `USE_SYSTEM_DATE` and without it (and it behaves as expected - uses find_package in the first case and fetches the library from github in the second one). Moreover, CI builds pass - it means that it works not only on my system :)

Regarding that:
> find_package would need to be in the root CMakeFile

In my practice I put find_package in different level CMakeLists.txt files (trying to minimize the scope and use them only where needed), and never have problems. Perhaps your information is correct for some old cmake versions, but now using `find_package` and `include` anywhere looks OK.

Reply:
Ah, sorry, my fault - everything was OK only with `-DBUILD_TESTING=OFF` (likely because sqlpp11 is header-only). You are right - `date::date` target is used in top-level CMakeLists.txt, that's why `find_package` should be located there too.